examples/headsail-bsp/src/lib.rs

```diff
@@ -4,6 +4,16 @@
 pub mod sprintln;
 pub mod uart;
 pub mod timer {
+    /*!
+     * Timer module for Headsail. When running on the Renode 
+     * Virtual Prototype, the "vp" feature should be enabled. 
+     * All operations are Read-Modify-Write.
+     * 
+     * HOW TO USE THIS DRIVER:
+     * In order to use any of the four timers that come with 
+     * Headsail HPC SubSystem, use the respective Timer{0..3} 
+     * type alias provided and the functions associated with it.
+     */
```

Comment:
Very nice.

Reply:
Please verify using `cargo doc --open` that this looks like how you intend it to look like. This outer attribute applies to the `timer` module and might compete with the outer attribute defined inside the module file.

Reply:
Have already tested it, this is the comment that is shown in the documentation pages. As far as the outer comments in the respective driver files, I am a bit lost on why they are not showing in the documentation, as I have already marked them as outer comments. I think I need your help in that as I am probably missing something here.

Reply:
I made a mistake with outer/inner; they're the other way around. I'll use the right terminology going forward.

I think this inner attribute here applies to the `mod timer` and supersedes the inner attribute specified inside the module (which also applies to `mod timer` and that might be why you're not seeing it.

Reply:
Then in that case, by removing the comment in the lib.rs file would result in the comment inside the timer_unit.rs to show, but this is not the case I see when I run `cargo doc --open -Fvp`.

Reply:
Alright, I'll investigate this to be able to give you a full answer. I won't hold up this merge though, so feel free to merge as the rest of the issues are done. Let's talk about this next week!

Reply:
I tried the documentation generation. Apparently rustdoc always takes the module documentation from the first definition at `lib.rs` even if it's empty. Therefore the in-file module-level documentation never makes it to the docs, as it's overridden by the re-export trickery.